File: Searching Algorithms/Ordered Linear Search/ordered-search.java
import java.util.Scanner;
public class ordered_linear{
    public static void main(String[] args){
        System.out.print('\u000C'); //Clear Screen Command in Java
 
        int n,i,flag=0,element,pos=-1;
        Scanner S=new Scanner(System.in);
        System.out.println("Enter the number of elements that has to be searched (they should be in sorted order):");
        n=S.nextInt();
        int arr[]=new int[n];
        System.out.println("Enter the elements in the array:");
        for(i=0;i<n;i++){
            arr[i]=S.nextInt();
        }
        System.out.println("Enter the element that has to be searched in the array:");
        element=S.nextInt();
        for(i=0;i<n;i++){
            if(arr[i]==element){
                flag=1;
                pos=i+1;
                break;
            }
            if(arr[i]>element){
                break;
            }
        }
        if(flag==1){
            System.out.println("Element exists in the array at position "+pos);
        }
        else{
            System.out.println("Element does not exist in the array");
        }
    }
}
        
            

File: Searching Algorithms/Ordered Linear Search/ordered-search.c
#include <stdio.h>
#include <stdlib.h>
int main(){
  int arr[100],n,i,pos=-1,element,flag=0;
  printf("Enter the number of elements you wish to enter: \n");
  scanf("%d",&n);
  printf("Enter the elements which should be in sorted order \n");
  for(i=0;i<n;i++){
    scanf("%d",&arr[i]);
  }
  printf("Enter the element to be searched in the array: \n");
  scanf("%d",&element);
  for(i=0;i<n;i++){
    if(arr[i]==element){
      pos=i+1;
      flag=1;
      break;
    }
    if(element<arr[i]){
      break;
    }
  }
  if(flag==1){
    printf("The elemnent has been found at position %d", pos);
  }
  else{
    printf("The elemnent has not been found");
  }
return 0;
  
}
